ui/package.json

```diff
@@ -106,10 +106,12 @@
     "react-grid-layout": "^0.13.9",
     "react-onclickoutside": "^5.2.0",
     "react-redux": "^4.4.0",
-    "react-router": "^2.4.1",
+    "react-router": "^3.0.2",
```

Comment:
3.0 does change the API for the most part (nothing we are using), it mostly just removes deprecation warnings and successfully excises deprecations from the package: https://github.com/ReactTraining/react-router/issues/3336. On the other hand, 4.0 is a breaking upgrade, and I felt that made it not worth the refactoring effort at this time.